os/arch/arm/src/armv7-a/arm_cpuhotplug.c

```diff
@@ -83,24 +86,41 @@ int arm_hotplug_handler(int irq, void *context, void *arg)
 			   (ie. TizenRT Kernel logic is not ready to handle such case yet)
 			   Thus, we will be facing a deadlock when cpuA fire SGI2 to cpuB (ie. pausing CPU)
 	*/
+	int cpu = this_cpu();
+	char curr_task[CORE_IDLE_TASK_CHAR_COUNT];
+	snprintf(curr_task, CORE_IDLE_TASK_CHAR_COUNT, "CPU%d IDLE", cpu);
+	/* If current cpu is in idle thread, then proceed for hotplug operation */
+	if (!strncmp(g_running_tasks[cpu]->name, curr_task, CORE_IDLE_TASK_CHAR_COUNT)) {
+		/* Set secondary core to hotplug state */
+		up_set_secondary_cpu_state(cpu, CPU_HOTPLUG);
 
-	/* Set secondary core to hotplug state */
-	up_set_secondary_cpu_state(this_cpu(), CPU_HOTPLUG);
-
-	/* Save the tcb state */
-	struct tcb_s *rtcb = this_task();
-	arm_savestate(rtcb->xcp.regs);
-	rtcb->task_state = TSTATE_TASK_ASSIGNED;
-	CURRENT_REGS = NULL;
+		/* Save the tcb state */
+		struct tcb_s *rtcb = this_task();
+		arm_savestate(rtcb->xcp.regs);
+		rtcb->task_state = TSTATE_TASK_ASSIGNED;
+		CURRENT_REGS = NULL;
 
-	/* MINOR: PSCI is a general interface for controlling power of cortex-A cores 
-	It is one of the feature in arm-trusted-firmware (ie. Trustzone-A)
-	Currently, the definitions are provided by chip specific arch layer
-	/* If the cpu off operation failed, we should face a problem during wakeup
-	booting the secondary core, thus we can ignore the return value here first
-	*/
-	/* Shut down the secondary core */
-   	(void)psci_cpu_off(); 
+		/* If SMP is enabled, the secondary core should take care of setting up the HW timer */
+		/* We need to add some flag to indicate the action has been completed, or else for 
+		   core number > 2, this code piece might be executed several times when each core goes hotplug,
+		   which might lead to some unexpected problem
+		*/
```

Comment:
Hi...
I don't understand why the second core needs to setup the hw timer in case of smp? Can you please elaborate?